src/mon/AuthMonitor.cc

```diff
@@ -1638,21 +1638,17 @@ bool AuthMonitor::prepare_command(MonOpRequestRef op)
     // auth get-or-create <name> [mon osdcapa osd osdcapb ...]
 
     if (!valid_caps(caps_vec, &ss)) {
-      err = -EINVAL;
-      goto done;
+      return -EINVAL;
```

Comment:
This method returns boolean, not integer:

```cpp
bool AuthMonitor::prepare_command(MonOpRequestRef op)
{
```

`return -EINVAL` changes the behavior the behavior in both the retvalue (`false` to `true`) and `reply_command()` aspects.

```cpp
done:
  rdata.append(ds);
  getline(ss, rs, '\0');
  mon.reply_command(op, err, rs, rdata, get_last_committed());
  return false;
}
```

I don't think it's intentional.

Reply:
Strange, I can't find this change.